process/block/sovereignChainBlock.go

```diff
@@ -792,6 +788,20 @@ func (scbp *sovereignChainBlockProcessor) ProcessBlock(headerHandler data.Header
 		return nil, nil, err
 	}
 
+	newBody := body
+	defer func() {
+		if err != nil {
+			return
+		}
+
+		if !check.IfNil(shardHeader) {
+			go getMetricsFromHeader(headerHandler.ShallowClone(), uint64(txCounts.GetTotal()), scbp.marshalizer, scbp.appStatusHandler)
```

Comment:
shardHeader in the go func ?

how can you have nil shardheader at this point ?

Even the body should not be nil.

Reply:
Right, we can get rid of those.
Fixed as suggested